README.md

```diff
@@ -38,6 +38,93 @@ deactivate
 ```
 
 
+# Configuration
+
+
+## Main configuration file
+
+The main configuration file uses the TOML format. The default file name is `.pyodk_config.toml`, and the default location is the user home directory. The file name and location can be customised by setting the environment variable `PYODK_CONFIG_FILE` to some other file path. The expected file structure is as follows:
+
+```
+[central]
+base_url = "https://www.example.com"
+username = "my_user"
+password = "my_password"
+default_project_id = 123
+```
+
+
+## Session cache file
+
+The session cache file uses the TOML format. THe default file name is `.pyodk_cache.toml`, and the default location is the user home directory. The file name and location can be customised by setting the environment variable `PYODK_CACHE_FILE` to some other file path. This file should not be pre-configured as it is used to store a session token after login.
+
+
+# Usage
+
+
+## Example
+
+```python
+from pyodk.client import Client
+
+
+with Client() as client:
+    projects = client.projects.read_all()
+    forms = client.forms.read_all()
+    form_id = next(forms).xmlFormId
+    submissions = client.submissions.read_all(form_id=form_id)
+    odata = client.odata.read_table(form_id=form_id)
+```
+
+The `Client` is not specific to a project, but a default `project_id` can be set by:
+
+- A `default_project_id` in the configuration file.
+- An initialisation argument on the client: `Client(project_id=1)`.
+- A property on the client: `client.project_id = 1`.
+
+
+## Endpoints
+
+Available endpoints on `Client`:
+
+- Projects
+  - read
```

Comment:
How did you pick this `read`/`read_all` convention (e.g. is it common in similar Python libs)? How do you imagine it extending to creation? How do you see these calls interacting with the dynamic query builder we've discussed?

I'm not sure I would have separated Submissions and OData in this custom part of the API (vs. the dynamic query builder). I don't think "OData" is particularly meaningful to users -- it's just a means to get a json representation of their data. Do you think that separation is important? The OData call could be something like `client.submissions.get_json`

I'm asking mostly to have a record of the design decisions so we can continue being consistent!

Reply:
> How did you pick this read/read_all convention

It's following CRUD which is a plain and protocol-agnostic way to refer to manipulating objects. Any of the below may be IO targets for pyodk (e.g. cache or helpers to SQL [sqlite or postgres] or files).

| CRUD | HTTP | SQL | Files |
| - | - | - | - |
| create | post | insert | write |
| read | get | select | read |
| update | put/patch | update | write/append |
| delete | delete | delete | delete |

The `all` suffix is in preference to 1) a 5th verb `list`, or 2) a suffix `many` which implies specifying more than one ID to read (but it accepts no parameters and returns all IDs). Specific formats get a suffix too e.g. `read_xml`, `read_all_table`. I've started mapping this out in the file at `docs/api_index.csv`.

>  separated Submissions and OData in this custom part of the API

I agree the name `OData` isn't very obvious, but I used it to follow the central docs. The exercise of making the API index doc helped clarify that these fit well under `forms` and `submissions` so I've moved them and deleted the `odata` endpoint.

Reply:
I'd like to propose a more freeform and descriptive approach to naming. With this part of the library's API we're not aiming for 1:1 parity with Central's API or full coverage so I think we should aim for as descriptive of names as we can and to also carefully design what parameters can be passed in.

The guiding question I'd like to use for naming is something like 

> How do people talk about the action that’s being performed? How do ODK docs and Central frontend talk about it?

E.g. we have the concept of “submission edits” so use client.submissions.edit rather than update. In the case of these reads, I think someone would say something like "ok, now I need to get this form's metadata". That leads me to `client.forms.get_metadata(projectid)` and `client.forms.get_metadata(projectid, formid)`.

This we mean we value expressiveness and consistency with ODK concepts over consistency between method names.

Reply:
Fine by me to get this merged with the current names and then to focus on naming in another PR!

Reply:
> I think someone would say something like "ok, now I need to get this form's metadata".

Right, but before that I think they'd say "let me list all of the forms in this project". So maybe it's `client.forms.list(projectid)` and `client.forms.get_metadata(projectid, formid, extended=False)`.

I know I just said I'd value consistency between method names less than other things but it would be nice if this were consistent across resources. I think the following work equally well:

- `client.submissions.list(projectid, formid, extended=False)`, `client.submissions.get_metadata(projectid, formid, extended=False)`
- `client.app_users.list(projectid, extended=False)` (no way to get metadata for a single one)

And wonderfully enough, that's exactly what the API docs say: "listing all xyz" and "getting xyz metadata".